### PR TITLE
Add sqrt LR scaling

### LINKED ISSUE #3: [Task4]: Implement sqrt learning rate scaling in VISSL
**Task**: Currently VISSL supports automatic scaling of a parameter (learning rate) as described https://github.com/facebookresearch/vissl/blob/a5c973bce3cd778a17a0045dd3ee2f56ef247624/docs/source/vissl_modules/optimizer.rst#auto-scaling-of-learning-rate . The `scaled_lr = ((batchsize_per_gpu * world_size) * base_value ) / base_lr_batch_size` is the scaling of the value `base_value` by the factor `(batchsize_per_gpu * world_size) / base_lr_batch_size`. Let's call this factor `scale_factor`. then the scaling becomes `scaled_lr = scale_factor * base_value`.

Our goal in this task is to extend the scaling to also support `sqrt` scaling i.e. we also want to be able to do `scaled_lr = sqrt(scale_factor) * base_value`. In terms of code in VISSL, this logic is implemented here https://github.com/facebookresearch/vissl/blob/master/vissl/utils/hydra_config.py#L288-L292 and https://github.com/facebookresearch/vissl/blob/master/vissl/utils/hydra_config.py#L310-L314

**Steps/Tips**:
1. Add the parameter key-value `scaling_type: linear` to the dictionary `https://github.com/facebookresearch/vissl/blob/master/vissl/config/defaults.yaml#L688` 
2. Add the if-else logic to computing the `scale_factor` based on whether `scaling_type=linear` or `scaling_type=sqrt`. The logic should be extended here https://github.com/facebookresearch/vissl/blob/master/vissl/utils/hydra_config.py#L288-L292 and https://github.com/facebookresearch/vissl/blob/master/vissl/utils/hydra_config.py#L310-L314
3. Add a unit test here https://github.com/facebookresearch/vissl/blob/master/tests/test_configs.py to check that the logic works correctly and returns the expected scaled_value.  Follow the test https://github.com/facebookresearch/vissl/blob/master/tests/test_configs.py#L74-L95 to check for a particular value of the learning rate. Please contact Priya if this step is not clear.
4. Modify the docstring https://github.com/facebookresearch/vissl/blob/master/vissl/utils/hydra_config.py#L192-L215 to extend it for `sqrt` as well
5. Modify the docs https://github.com/facebookresearch/vissl/blob/master/docs/source/vissl_modules/optimizer.rst#auto-scaling-of-learning-rate to extend it to `sqrt` as well
6. Modify the description in the `defaults.yaml` here https://github.com/facebookresearch/vissl/blob/master/vissl/config/defaults.yaml#L687 to extend it to `sqrt`. Simply add one line indicating that the `sqrt` is also handled.
7. Send the PR to vissl review and tag Priya as reviewer

### PR DESCRIPTION
**Pull request by akainth015**
Resolves #3

### CODE REVIEW

File: vissl/utils/hydra_config.py [hunk 1]
```diff
@@ -285,7 +288,12 @@ def infer_learning_rate(cfg):
         param_schedulers = cfg.OPTIMIZER.param_schedulers.lr
         base_lr = param_schedulers.auto_lr_scaling.base_value
         base_lr_batch_size = param_schedulers.auto_lr_scaling.base_lr_batch_size
+        scaling_type = param_schedulers.auto_lr_scaling.scaling_type
+
         scale_factor = float(batch_size) / base_lr_batch_size
+        print(scale_factor)
```
Comment (prigoyal) [inline vissl/utils/hydra_config.py]: let's remove the print statement

File: vissl/utils/hydra_config.py [hunk 2]
```diff
@@ -285,7 +288,12 @@ def infer_learning_rate(cfg):
         param_schedulers = cfg.OPTIMIZER.param_schedulers.lr
         base_lr = param_schedulers.auto_lr_scaling.base_value
         base_lr_batch_size = param_schedulers.auto_lr_scaling.base_lr_batch_size
+        scaling_type = param_schedulers.auto_lr_scaling.scaling_type
+
```
Comment (prigoyal) [inline vissl/utils/hydra_config.py:296]: can we add an `assert scaling_type in ['sqrt', 'linear'], "Only linear | sqrt scaling_types are supported"`

File: tests/test_configs.py [hunk 3]
```diff
@@ -95,6 +95,23 @@ def test_cfg_cli_composition(self):
         )
 
 
+class TestScalingTypeConfig(unittest.TestCase):
```
Comment (prigoyal) [inline tests/test_configs.py:98]: very nice! <3

**Review by prigoyal — COMMENTED** [review]
looks great , small comments and good to go otherwise!

**Review by prigoyal — APPROVED** [review]
Looks AWESOME!!! thank you @akainth015 and @grace-omotoso :)
